Jenkinsfile

```diff
@@ -0,0 +1,56 @@
+#!groovy
+
+node ('docker') {
+    stage 'init'
+    //deleteDir()
+    checkout scm
+
+    // Run selenium in a docker container of its own on the host.
+    // It will output the selenium server address to ./target/.selenium_ip
+    sh "./start-selenium.sh"
+    def seleniumIP = readFile './target/.selenium_ip'
+
+    try {
+        // Build an image from the the local Dockerfile
+        def athImg = docker.build('blueocean-ath-builder')
+
+        // Expose the port on which the ATH Jenkins instance runs (12345), allowing the
+        // Firefox browser (running in the selenium container) to make requests back
+        // in etc.
+        //
+        // To bind in the local ~/.m2 when running in dev mode, simply add the following
+        // volume binding to the "inside" container run settings (change username from "tfennelly"):
+        //       -v /home/tfennelly/.m2:/home/bouser/.m2
+        //
+        athImg.inside("--expose=12345 -p 12345:12345") {
+            withEnv(["BLUEO_SELENIUM_SERVER_ADDR=${seleniumIP}"]) {
+                try {
+                    sh "echo 'Starting build stage'"
+                    // Build blueocean and the ATH
+                    stage 'build'
+                    dir('blueocean-plugin') {
```

Comment:
couldn't we use a docker image instead to check out and build?

Further this checks put always master, right? Wouldn't we need as well the ability to fetch branches?